docs/mp/openapi/01_openapi.adoc

```diff
@@ -182,15 +160,241 @@ include::{common-openapi-doc}[common-config]
+import org.eclipse.microprofile.openapi.models.PathItem;
+import org.eclipse.microprofile.openapi.models.Paths;
+
+/**
+ * Defines two paths using the OpenAPI model reader mechanism, one that should
+ * be suppressed by the filter class and one that should appear in the published
+ * OpenAPI document.
+ */
+public class SimpleAPIModelReader implements OASModelReader {
+
+    /**
+     * Path for the example endpoint added by this model reader that should be visible.
+     */
+    public static final String MODEL_READER_PATH = "/test/newpath";
+
+    /**
+     * Path for an endpoint that the filter should hide.
+     */
+    public static final String DOOMED_PATH = "/test/doomed";
+
+    /**
+     * ID for an endpoint that the filter should hide.
+     */
+    public static final String DOOMED_OPERATION_ID = "doomedPath";
+
+    /**
+     * Summary text for the endpoint.
+     */
+    public static final String SUMMARY = "A sample test endpoint from ModelReader";
+
+    @Override
+    public OpenAPI buildModel() {
+        /*
+         * Add two path items, one of which we expect to be removed by
+         * the filter and a very simple one that will appear in the
+         * published OpenAPI document.
+         */
+        PathItem newPathItem = OASFactory.createPathItem()
+                .GET(OASFactory.createOperation()
+                    .operationId("newPath")
+                    .summary(SUMMARY));
+        PathItem doomedPathItem = OASFactory.createPathItem()
+                .GET(OASFactory.createOperation()
+                    .operationId(DOOMED_OPERATION_ID)
+                    .summary("This should become invisible"));
+        OpenAPI openAPI = OASFactory.createOpenAPI();
+        Paths paths = OASFactory.createPaths()
+                .addPathItem(MODEL_READER_PATH, newPathItem)
+                .addPathItem(DOOMED_PATH, doomedPathItem);
+        openAPI.paths(paths);
+
+        return openAPI;
+    }
+}
+----
+
+The classes with filters should be specified in `META-INF/microprofile-config.properties`:
+
+[source,properties]
+----
+mp.openapi.filter=io.helidon.microprofile.examples.openapi.basic.internal.SimpleAPIFilter
+mp.openapi.model.reader=io.helidon.microprofile.examples.openapi.basic.internal.SimpleAPIModelReader
+----
+
+
+Now just build and run
+
+[source,bash]
+----
+mvn package
+java -jar target/helidon-examples-microprofile-openapi-basic.jar
+----
+
+Try the endpoints:
+
+[source,bash]
+----
+curl -X GET http://localhost:8080/greet
+{"message":"Hello World!"}
+
+curl -X GET http://localhost:8080/openapi
+[lengthy OpenAPI document]
+----
+
+The output describes not only then endpoints from `GreetResource` but
+also one contributed by the `SimpleAPIModelReader`.
+
+Full example is available link:https://github.com/oracle/helidon/tree/{health-release}/examples/microprofile/openapi-basic[in our official repository]
+
+
+== Additional information
+
+Jandex usage Highly recommended.
+
+==== Building the Jandex index
```

Comment:
I think we can remove the introductory line. Reduce the header level by one and change the header to "Using a Jandex Index" or maybe adding "(highly recommended)" to the end of the header.

Reply:
Fixed)

Reply:
Please reduce the header level by one on "Building the Jandex index"

Reply:
Did!